### PR TITLE
fix(api-graphql): custom operation selection set on nested custom types

### PR DESCRIPTION
**Pull request by HuiSF**
#### Description of changes

* Updated custom operation impl. to use pre-existing selection set generation utility function so it can handle nested custom types
* Added unit tests against the nested custom types scenarios
* Bump data schema packages versions to gain new implementations

#### Description of how you validated changes

1. unit tests
2. manual E2E tests in a sample app

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/api-graphql/src/internals/operations/custom.ts [hunk 1]
```diff
@@ -251,11 +233,12 @@ function operationSelectionSet(
 	) {
 		return '';
 	} else if (hasStringField(operation.type, 'nonModel')) {
-		const model = modelIntrospection.nonModels[operation.type.nonModel];
-		return `{${modelishTypeSelectionSet(model)}}`;
+		const nonModel = modelIntrospection.nonModels[operation.type.nonModel];
+		return `{${selectionSetIRToString(
```
Comment (stocaaro) [inline packages/api-graphql/src/internals/operations/custom.ts:237]: We're wrapping strings in surrounding curly brackets, `{}`, in several places. I find the character sequence `` `{${`` a bit hard to read, which makes me wonder if a few extra chars for `wrapInCurlyBrackets(x)` might be more readable.

Not specific to this PR, but a thing I thought of.

**Review by stocaaro — APPROVED** [review]
Makes sense. I think all of the below are more nit than concern. Good to see this ship as is or with any attention to these comments.

**Review by iartemiev — APPROVED** [review]
LGTM. 
As Aaron suggested, would be nice to import from a top-level `utils/index.ts` instead of sub-dirs, but that's non-blocking

**Comment by HuiSF** [comment]
Thanks if this is non-blocking I will open a separate PR to refactor further (also wanted to extra the selection set utils out from that long source file)

**Review by svidgen — APPROVED** [review]

**Pull request merged**